benchmark/Support/autoplot.jl

```diff
@@ -0,0 +1,261 @@
+using BenchmarkTools
+using Catlab, Catlab.Theories
+using Catlab.CategoricalAlgebra
+using Catlab.Graphs
+using Catlab.Graphics
+using Catlab.Graphics.Graphviz
+using Colors
+using Plots
+
+# add_loops
+add_loops!(g) = add_parts!(g, :E, nparts(g, :V), src=parts(g, :V), tgt=parts(g, :V))
+add_loops(g) = begin
+    h = copy(g)
+    add_loops!(h)
+    return h
+end
+
+# Auto Calculation and Plotting
+# give a list of to and from graphs -> tests and plots them based on vertex/edge amounts
+function autoPlot(fromList, toList)
+    if length(fromList) != length(toList)
+        println("From and to lists should be the same size.")
+        return
+    end
+    println("Autoplotting Homs...\nTotal pairs: $(length(toList))")
+    x1 = Int64[]
+    y1 = Float64[]
+    x2 = Int64[]
+    y2 = Float64[]
+    x3 = Int64[]
+    x4 = Int64[]
+    for i in 1:length(fromList)
+        print("Graph pair $i:   ")
+        f = fromList[i]
+        t = toList[i]
+        #injection
+        append!(y1, time(median(@benchmark homomorphism($f, add_loops($t)))))
+        append!(x1, length(vertices(f)))
+        append!(x3, length(edges(t)))
+        print("Injection complete.   ")
+        #surjection
+        append!(y2, time(median(@benchmark homomorphism($t, add_loops($f)))))
+        append!(x2, length(vertices(t)))
+        append!(x4, length(edges(t)))
+        print("Surjection complete.\n")
+    end
+    scatter([x1, x2], [y1, y2], title="Autoplotted Graph Vertices", xlabel="Number of \"From\" Vertices", ylabel="Single Hom Calculation Time (ns)", label=["Injective" "Surjective"])
+    savefig("autoVertex.png")
+    scatter([x3, x4], [y1, y2], title="Autoplotted Graph Edges", xlabel="Number of Edges", ylabel="Single Hom Calculation Time (ns)", label=["Injective" "Surjective"])
+    savefig("autoEdge.png")
+end
+
+# Less accurate... Much faster.
+function quickPlot(fromList, toList)
```

Comment:
Is this faster because you only look at the zipped list of graphs to benchmark, rather than the Cartesian product of the lists? I would rather see this refractored to use `Base.Iterators.zip` and `Base.Iterators.product` and have an interface for just taking a sequence of pairs of graphs and running the benchmarks and making the scatter plot. Then the sequence of pairs of graphs could be created with different Iterators

Reply:
For some reason I had forgotten those functions were still in that file. I need to adjust what I have in there since the plotting functions were used in the old plots I made toward the beginning of the semester.

That being said, I will work on them.